python-package/xgboost/_typing.py

```diff
@@ -1,20 +1,24 @@
 """Shared typing definition."""
 import ctypes
 import os
-from typing import Optional, List, Any, TypeVar, Union
+from typing import Optional, List, Any, TypeVar, Union, Type
 
 # os.PathLike/string/numpy.array/scipy.sparse/pd.DataFrame/dt.Frame/
 # cudf.DataFrame/cupy.array/dlpack
+import numpy as np
+
 DataType = Any
 
 # xgboost accepts some other possible types in practice due to historical reason, which is
 # lesser tested.  For now we encourage users to pass a simple list of string.
 FeatureNames = Optional[List[str]]
+FeatureTypes = Optional[List[str]]
```

Comment:
We might want to use Sequence instead.

Reply:
I think it's feasible and am able to pass mypy tests, although internally both `FeatureNames` and `FeatureTypes` are `list`s.

Reply:
Thank you for making the change!